Documentation/storage.md

```diff
@@ -147,6 +147,32 @@ storage:
 
 Dex determines the namespace it's running in by parsing the service account token automatically mounted into its pod.
 
+## Migrating from TPRs to CRDs
+
+This section descibes how users can migrate storage data in dex when upgrading from an older version of kubernetes (lower than 1.7). This involves creating new CRDs and moving over the data from TPRs.
+The flow of the migration process is as follows:
+1. Stop running old version of Dex (lower than v2.7.0).
+2. Create new CRDs by running the following command:
+   ```
+   kubectl create -f scripts/manifests/crds/
+   ```
+   Note that the newly created CRDs have `dex.coreos.com` as their group and will not conflict with the existing TPR resources which have `oidc.coreos.com` as the group.
+3. Migrate data from existing TPRs to CRDs by running the following commands for each of the TPRs:
+   1. Use the following command to store TPR definition in a temporary yaml file:
+      ````
+      kubectl get --namespace=tectonic-system <TPR-name>.oidc.coreos.com  -o yaml >> out.yaml
```

Comment:
Can you put this in a for loop with all of the resources?

Reply:
Would you rather have me put all the instructions into the for loop?

Reply:
Something like

```
DEX_NAMESPACE=dex-system

for RESOURCE in authcodes connectors oauth2clients offlinesessionses; do
     kubectl get --namespace=$DEX_NAMESPACE $RESOURCE.oidc.coreos.com  -o yaml >> out.yaml
done
```

Reply:
Put this into a script. If you'd like I could put all the instructions into that script...but detailed them out so the user knows what they are doing.